### ISSUE TITLE
Legion: Allow replay of non-idempotent traces

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#1061]
Currently the runtime does not allow replaying non-idempotent traces this restriction should be relaxed. 

Second, an interface should be added to the mapper to allow copies to be issued when replaying traces fail.

**Comment by lightsighter** [comment]
We're definitely going to do the first one.

The second one is still on the fence and I think we'll need more evidence that it is useful in order to actually do it.

**Comment by lightsighter** [comment]
Non-idempotent traces are now supported in the master branch of Legion (and have been for quite some time). I don't see much need for the second feature request now that we do support non-idempotent traces; either a template's preconditions will match and we'll replay or we won't find a matching template and we capture. If there is any steady-state to the execution eventually we'll start getting templates that match. There's no need to issue extra "fixup" copies to make things work.
